crates/agglayer-node/src/lib.rs

```diff
@@ -27,7 +27,7 @@ use agglayer_telemetry::ServerBuilder as MetricsBuilder;
 ///
 /// This function returns on fatal error or after graceful shutdown has
 /// completed.
-pub fn main(cfg: PathBuf) -> Result<()> {
+pub fn main(cfg: PathBuf, version: &str) -> Result<()> {
```

Comment:
Note: to read cargo env variables `build.rs` has to be executed for the particular crate. Felt like overkill to make separate `build.rs` for `agglayer`,  `agglayer-prover` and `agglayer-node`, so I chose to pass the global `agglayer` package version info instead.